tq/meter.go

```diff
@@ -141,12 +142,29 @@ func (m *Meter) StartTransfer(name string) {
 
 // TransferBytes increments the number of bytes transferred
 func (m *Meter) TransferBytes(direction, name string, read, total int64, current int) {
+	now := time.Now()
+
```

Comment:
I'd put this under the `if m == nil` check, since nothing happens with `now` before that return.

Reply:
Good call. I was wavering on this when writing the patch, and left it at the top because I had convinced myself that waiting until _after_ the `if` would introduce imprecision in the averaging. While probably true to a degree, I'm sure that the effect is negligible, and the code would be more readable if the "first thing" that is supposed to happen came after the nullity check. Changed in cf8d827.